java/vector/src/main/java/org/apache/arrow/vector/extension/UnknownType.java

```diff
@@ -0,0 +1,379 @@
+  }
+
+  @Override
+  public String toString() {
+    return "UnknownType("
+        + storageType
+        + ", typeName='"
+        + typeName
+        + '\''
+        + ", vendorName='"
+        + vendorName
+        + '\''
+        + ')';
+  }
+
+  private static class UnderlyingVectorTypeVisitor implements ArrowTypeVisitor<FieldVector> {
+    private final String name;
+    private final BufferAllocator allocator;
+
+    UnderlyingVectorTypeVisitor(String name, BufferAllocator allocator) {
+      this.name = name;
+      this.allocator = allocator;
+    }
+
+    @Override
+    public FieldVector visit(Null type) {
+      return new NullVector(name);
+    }
+
+    private RuntimeException unsupported(ArrowType type) {
+      throw new UnsupportedOperationException(
+          "UnknownType#getUnderlyingVector is not supported for storage type: " + type);
+    }
+
+    @Override
+    public FieldVector visit(Struct type) {
+      throw unsupported(type);
+    }
+
+    @Override
+    public FieldVector visit(List type) {
+      throw unsupported(type);
+    }
+
+    @Override
+    public FieldVector visit(LargeList type) {
+      throw unsupported(type);
+    }
+
+    @Override
+    public FieldVector visit(FixedSizeList type) {
+      throw unsupported(type);
+    }
+
+    @Override
+    public FieldVector visit(Union type) {
+      throw unsupported(type);
+    }
+
+    @Override
+    public FieldVector visit(Map type) {
+      throw unsupported(type);
+    }
+
+    @Override
+    public FieldVector visit(Int type) {
+      return new IntVector(name, allocator);
+    }
+
+    @Override
+    public FieldVector visit(FloatingPoint type) {
+      switch (type.getPrecision()) {
+        case HALF:
+          return new Float2Vector(name, allocator);
+        case SINGLE:
+          return new Float4Vector(name, allocator);
+        case DOUBLE:
+          return new Float8Vector(name, allocator);
+      }
+      throw unsupported(type);
+    }
+
+    @Override
+    public FieldVector visit(Utf8 type) {
+      return new VarCharVector(name, allocator);
+    }
+
+    @Override
+    public FieldVector visit(Utf8View type) {
+      return new ViewVarCharVector(name, allocator);
+    }
+
+    @Override
+    public FieldVector visit(LargeUtf8 type) {
+      return new LargeVarCharVector(name, allocator);
+    }
+
+    @Override
+    public FieldVector visit(Binary type) {
+      return new VarBinaryVector(name, allocator);
+    }
+
+    @Override
+    public FieldVector visit(BinaryView type) {
+      return new ViewVarBinaryVector(name, allocator);
+    }
+
+    @Override
+    public FieldVector visit(LargeBinary type) {
+      return new LargeVarBinaryVector(name, allocator);
+    }
+
+    @Override
+    public FieldVector visit(FixedSizeBinary type) {
+      return new FixedSizeBinaryVector(Field.nullable(name, type), allocator);
+    }
+
+    @Override
+    public FieldVector visit(Bool type) {
+      return null;
```

Comment:
Isn't boolean supposed to use a Fixed Size Primitive Layout? Just for me to learn, why there's not a `BoolVector` or a `FixedSizeBoolVector`?

Reply:
This is incomplete